onnxruntime/test/fuzzing/src/OnnxPrediction.cpp

```diff
@@ -2,40 +2,57 @@
 // Licensed under the MIT License.
 
 #include "OnnxPrediction.h"
+#include <vector>
+#include <memory>
 
 // Uses the onnxruntime to load the model
 // into a session.
 //
 OnnxPrediction::OnnxPrediction(std::wstring& onnx_model_file)
-:
-rawModel{nullptr},
-ptrSession{nullptr},
-session{ env, onnx_model_file.c_str(), emptySessionOption},
-inputNames{session.GetInputCount()},
-outputNames{session.GetOutputCount()}
-{
+    : rawModel{nullptr},
+      ptrSession{nullptr},
+      session{env, onnx_model_file.c_str(), emptySessionOption},
+      inputNames{session.GetInputCount()},
+      outputNames{session.GetOutputCount()} {
   init();
 }
 
 // Uses the onnx to seri
 //
 OnnxPrediction::OnnxPrediction(onnx::ModelProto& onnx_model)
-:
-session{nullptr}
-{
-  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),   
-  [this](void * ptr)
-      { 
-      this->GetAllocator().Free(ptr); 
-      }
-  };
+    : session{nullptr} {
+  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),
+                                   [this](void* ptr) {
+                                     this->GetAllocator().Free(ptr);
+                                   }};
 
   onnx_model.SerializeToArray(rawModel.get(), static_cast<int>(onnx_model.ByteSizeLong()));
 
-  ptrSession = std::make_unique<Ort::Session>(env, 
-      rawModel.get(), 
-      onnx_model.ByteSizeLong(), 
-      emptySessionOption),
+  ptrSession = std::make_unique<Ort::Session>(env,
+                                              rawModel.get(),
+                                              onnx_model.ByteSizeLong(),
+                                              emptySessionOption),
+
+  inputNames.resize(ptrSession->GetInputCount());
+  outputNames.resize(ptrSession->GetOutputCount());
+
+  init();
+}
+
+OnnxPrediction::OnnxPrediction(const std::vector<char>& modelData)
+    : session{nullptr} {
+  size_t numBytes = modelData.size();
+  rawModel = std::shared_ptr<void>{alloc.Alloc(numBytes),
+                                   [this](void* ptr) {
```

Comment:
shared_ptr should rarely be needed. avoid wherever possible.

Reply:
Due to the custom destructor, I could not make it a unique_ptr. We want to have this memory managed i guess instead of freeing memory in the the destructor. Let me know if we want to make raw_model a raw pointer. 

---
In reply to: [521684570](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521684570) [](ancestors = 521684570)